### ISSUE TITLE
Modal's aria-label[ledby] and aria-describedby

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/carbon#2270]
While looking at [a Carbon React Modal accessibility issue](https://github.com/IBM/carbon-components-react/issues/1932), I decided to compare the React Modal to the [Vanilla Modal](https://www.carbondesignsystem.com/components/modal/code). I noticed that the Vanilla Carbon Modal does connect up `aria-labelledby` and `aria-describedby`, however there are a couple of potential problems:
1. Apparently the Carbon `modalLabel` is optional? That is very strange. Is the `modalHeading` expected to be the dialog title, then? Usually, the text at the top of a dialog is considered to be the dialog name, i.e. "Find" dialog, "Open File" dialog, "Create New User" dialog, "Login" dialog. There must always be a valid `aria-label` or `aria-labelledby` on a dialog, so that screen readers can announce the dialog's name when it becomes visible. So please either make the Carbon `modalLabel` required and just hook it up automatically to `aria-labelledby`, or hook up the Carbon `modalHeading` to `aria-labelledby`.
**[Edit**: just noticed the `modalAriaLabel` prop. I guess you could document that one of `modalLabel` (visible dialog name) or `modalAriaLabel` (invisible dialog name) must be provided.**]**
2. Usually, `aria-describedby` points to the sentence or sentences that describe what the dialog is for, or how to use the dialog, i.e. like the "Lorem ipsum" text that Carbon has in the "bx--modal-content" element in the example. (It could also point to the `modalHeading` plus the text).

I would suggest that any redesign includes thinking about how a Modal's label and description could be marked up for accessibility. Note that:
a) `aria-label[ledby]` is not optional. If a Carbon Modal does not have a `modalLabel`, then `aria-labelledby` needs to point to the `modalHeading`. **[Edit**: or you could use `modalAriaLabel` if it was provided.**]**
b) `aria-describedby` is optional, but if there is [a heading and] some text that describes how to use the dialog, then `aria-describedby` should point to it.

**Comment by dakahn** [comment]
This is an important fix for sure. 👍🏽

**Comment by dakahn** [comment]
Sorry this went stale, but **still mega important.** Removing wontfix
